### PR TITLE
437 source reference groups

### PR DESCRIPTION
**Pull request by bonndan**
#437 groups and templates can be set in source references

### CODE REVIEW

File: src/test/java/de/bonndan/nivio/input/dto/InputFormatHandlerNivioTest.java [hunk 1]
```diff
@@ -87,16 +93,35 @@ public void readServiceAndInfra() {
...
-        assertEquals("Keycloak SSO", service.getName());
-        assertEquals("keycloak", service.getIdentifier());
+        //when
+        descriptionFactory.applyData(file, null, landscapeDescription);
+
+        //then
+        Map<String, GroupDescription> groups = landscapeDescription.getGroups();
+        assertThat(groups).isNotEmpty();
+
+        GroupDescription ingress = groups.get("ingress");
```
Comment (bonndan) [inline src/test/java/de/bonndan/nivio/input/dto/InputFormatHandlerNivioTest.java:109]: @mrgamedev07 here we test for a new group definition that comes from a referenced file

File: src/main/java/de/bonndan/nivio/input/SourceReferencesResolver.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void resolve(final LandscapeDescription landscapeDescription) {
             }
 
             try {
-                landscapeDescription.addItems(factory.getDescriptions(ref, baseUrl));
+                factory.applyData(ref, baseUrl, landscapeDescription);
```
Comment (mrgamedev07) [inline src/main/java/de/bonndan/nivio/input/SourceReferencesResolver.java:50]: variable name factory might be changed. It confuses with the spring bean factory.
 
``InputFormatHandler  inputFormatHandlerFactory;                                                              
inputFormatHandlerFactory.applyData(ref, baseUrl, landscapeDescription);``
This should be read able .i guess !!!

File: src/test/java/de/bonndan/nivio/input/KubernetesTest.java [hunk 3]
```diff
@@ -52,20 +53,26 @@ void setup() {
     }
 
     @Test
-    public void testRead() throws IOException {
+    public void testRead() {
```
Comment (mrgamedev07) [inline src/test/java/de/bonndan/nivio/input/KubernetesTest.java:56]: Just added a comment here, for confirmation .you have removed IOException here. are you sure?

File: src/test/java/de/bonndan/nivio/input/dto/InputFormatHandlerNivioTest.java [hunk 4]
```diff
@@ -16,29 +16,35 @@
 import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.*;
```
Comment (mrgamedev07) [inline src/test/java/de/bonndan/nivio/input/dto/InputFormatHandlerNivioTest.java:22]: Import * again ;

**Pull request merged**